core/git.go

```diff
@@ -705,43 +773,47 @@ func doGitCheckout(
 		pullref += ":" + pullref
 	}
 
+	// Fetch and check out the requested ref
 	args := []string{"fetch", "-u"}
 	if depth > 0 {
 		args = append(args, fmt.Sprintf("--depth=%d", depth))
 	}
 	args = append(args, "origin", pullref)
-	_, err = checkoutGit.Run(ctx, args...)
-	if err != nil {
+	if _, err = checkoutGit.Run(ctx, args...); err != nil {
 		return err
 	}
-	_, err = checkoutGit.Run(ctx, "checkout", strings.TrimPrefix(fullref, "refs/heads/"))
-	if err != nil {
+
+	if _, err = checkoutGit.Run(ctx, "checkout", strings.TrimPrefix(fullref, "refs/heads/")); err != nil {
 		return fmt.Errorf("failed to checkout remote %s: %w", remote, err)
 	}
-	_, err = checkoutGit.Run(ctx, "reset", "--hard", commit)
-	if err != nil {
+	if _, err = checkoutGit.Run(ctx, "reset", "--hard", commit); err != nil {
 		return fmt.Errorf("failed to reset ref %s: %w", remote, err)
 	}
-	_, err = checkoutGit.Run(ctx, "remote", "set-url", "origin", remote)
-	if err != nil {
+	if _, err = checkoutGit.Run(ctx, "remote", "set-url", "origin", remote); err != nil {
 		return fmt.Errorf("failed to set remote origin to %s: %w", remote, err)
 	}
-	_, err = checkoutGit.Run(ctx, "reflog", "expire", "--all", "--expire=now")
-	if err != nil {
+	if _, err = checkoutGit.Run(ctx, "reflog", "expire", "--all", "--expire=now"); err != nil {
 		return fmt.Errorf("failed to expire reflog for remote %s: %w", remote, err)
 	}
-
 	if err := os.Remove(filepath.Join(checkoutDirGit, "FETCH_HEAD")); err != nil && !errors.Is(err, os.ErrNotExist) {
 		return fmt.Errorf("failed to remove FETCH_HEAD for remote %s: %w", remote, err)
 	}
 
+	// Sync submodule URLs with .gitmodules so the next update pulls from the right location.
```

Comment:
not sure I follow why this is needed. What does the "so the next update" means? Isn't this `doGitCheckout` function only gets called for new checkouts each time? How would we need to do a `submodule sync`? 🤔

Reply:
Yeah, you’re right: this was a leftover from my earlier testing that I thought was needed.

In our flow, `doGitCheckout` is always working on a brand-new worktree, and the submodules are initialized for the first time via `git submodule update --init --recursive`. That means `.gitmodules` and `.git/config` are already in sync at that point, so running `git submodule sync` was redundant 👌